src/socket/waker_store.rs

```diff
@@ -0,0 +1,23 @@
+use core::task::Waker;
+
+#[derive(Debug)]
+pub struct WakerStore {
+    waker: Option<Waker>,
+}
+
+impl WakerStore {
+    pub const fn new() -> Self {
+        Self { waker: None }
+    }
+
+    pub fn register(&mut self, w: &Waker) {
+        match self.waker {
+            Some(ref w2) if (w2.will_wake(w)) => {}
```

Comment:
I think it's worth adding a comment here explaining this logic.

Reply:
Done!